### ISSUE TITLE
[feature request] make spray can read content from region

### ISSUE THREAD

**Issue by stardiviner** [ian-kelling/spray#7]
Why?

Because in many cases, the buffer content is not full of words which is easy to read.
for example:
```
This is a good day.

[[http://www.google][Google]]
```

I only want to read the part `This is a good day`.
Of course this is just a simple content example.

I try to separate your functions: `spray--word-at-point` and `spray--update` out a function called `spray-read`, but I failed.
I only can read your source code, but don't know how to modify it.

**Comment by ian-kelling** [comment]
I agree this would be good to have. When I have time, I will add a comment, until then, any contributions are welcome.
